components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -39,76 +38,73 @@ internal val additionalHeaders = mapOf(
  * WebView-based EngineSession implementation.
  */
 @Suppress("TooManyFunctions")
-class SystemEngineSession(private val defaultSettings: Settings? = null) : EngineSession() {
+class SystemEngineSession(
+    private val context: Context,
+    private val defaultSettings: Settings? = null
+) : EngineSession() {
 
-    internal var view: WeakReference<SystemEngineView>? = null
-    internal var scheduledLoad = ScheduledLoad(null)
+    @Volatile internal lateinit var internalSettings: Settings
     @Volatile internal var historyTrackingDelegate: HistoryTrackingDelegate? = null
     @Volatile internal var trackingProtectionPolicy: TrackingProtectionPolicy? = null
     @Volatile internal var webFontsEnabled = true
-    @Volatile internal var internalSettings: Settings? = null
+    @Volatile internal var currentUrl = ""
+    @Volatile internal var fullScreenCallback: WebChromeClient.CustomViewCallback? = null
+
+    // This is public for FFTV which needs access to the WebView instance. We can mark it internal once
+    // https://github.com/mozilla-mobile/android-components/issues/1616 is resolved.
+    @Volatile var webView: WebView = NestedWebView(context)
+        set(value) {
+            field = value
+            initSettings()
+        }
```

Comment:
Does this need to be `var`? Would it be possible to only expose a `val` outside of the component (for FFTV?). Can we add a ⚠️☢️☣️💥🧨 warning KDoc that this property is only a workaround and should not be used? :)

Reply:
Yeah, we need this mutable for the window request support, but we can hide it (make the setter internal). Updated!